src/fromager/sources.py

```diff
@@ -204,6 +204,17 @@ def download_git_source(
     if url_to_clone.startswith("git+"):
         url_to_clone = url_to_clone[len("git+") :]
 
+    # Parse ref from URL if it's in the form repo@ref
+    from urllib.parse import urlparse
+
+    if ref is None:
```

Comment:
Where do we have a call that's cloning from a URL where the caller does not pass a valid ref?